### PR TITLE
Renamed Blog to News.

### PR DESCRIPTION
**Pull request by toby63**
Signed-off-by: Tobias Gerold <tobias@g3ro.eu>

### CODE REVIEW

**Comment by toby63** [comment]
Needs to be tested, I guess.

File: hugo/config.toml [hunk 1]
```diff
@@ -6,7 +6,7 @@ publishDir = "../public"
 disableKinds = ["taxonomy","taxonomyTerm"]
 
 [permalinks]
-  blog = "/blog/:title/"
```
Comment (Krzmbrzl) [inline hugo/config.toml:9]: We should have a redirect in case someone has hardcoded the old name somewhere

File: hugo/content/news/2010-08-17-for-the-record.md [hunk 2]
```diff
@@ -20,18 +20,18 @@ Let&#8217;s start with the basic rules we established for the recorder:
 * For privacy reasons we will not allow recording on pre 1.2.3 servers. (They are not able to properly warn their users)
 * There will be no ACLs to disallow it on newer servers. (We feel that if we restrict the feature that way the Open Source nature of our project will lead to forks with the warnings completely removed which is an even worse situation for the users)
 
-Once you are on a server that fulfils these prerequisites you can open the Recorder via our Icon toolbar or the menu (Self->Recorder) and &#8220;Start&#8221; the recording. You can record for as long as your disk space lasts or until you grow tired of it. The data is encoded and written away on the fly. Supported audio formats at this time are:
+Once you are on a server that fulfils these prerequisites you can open the Recorder via our Icon toolbar or the menu (Self->Recorder) and &#8220;Start&#8221; the recording. You can record for as long as your disk space lasts or until you grow tired of it. The data is encoded and written away on the fly. Supported audio formats at this time are:
```
Comment (Krzmbrzl) [inline hugo/content/news/2010-08-17-for-the-record.md:23]: What is this kind of change about? Seems unrelated

File: hugo/content/security/Mumble-SA-2014-005.md [hunk 3]
```diff
@@ -1,63 +1,63 @@
...
-
-The fix for Mumble 1.2.6 addresses this issue by removing the Mumble client's ability to render SVG files received from external sources. The binary releases of the Mumble client for Windows and OS X also contain a patched version of Qt that removes the code paths in QSvg that can trigger local file access.
-
-A patch that applies on top of Mumble 1.2.5 can be found at http://mumble.info/security/Mumble-SA-2014-005.patch (for a detached PGP signature, append '.sig').
-
-The fix is also available in the Mumble git repo as c7aecb2956f214cd83b7a862a4fcf15cc76c4450 (on the v1.2.6 branch).
-
-The patch has been applied to Mumble's Qt tree is available in the mumble-developers-qt tre as
-	2147fa767980fe27a14f018b1528dbf880b96814.
+---
```
Comment (Krzmbrzl) [inline hugo/content/security/Mumble-SA-2014-005.md:1]: What happened here?

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]
There seem to be quite a few unrelated changes in here

File: hugo/content/security/Mumble-SA-2014-005.md (hunk 3 above)
Comment (toby63) [reply hugo/content/security/Mumble-SA-2014-005.md:1]: What exactly do you mean?

File: hugo/content/news/2010-08-17-for-the-record.md (hunk 2 above)
Comment (toby63) [reply hugo/content/news/2010-08-17-for-the-record.md:23]: I have no clue, but it seems that spaces were added or not?

File: hugo/config.toml (hunk 1 above)
Comment (toby63) [reply hugo/config.toml:9]: Good idea. 
:thinking: It looks like we could just use:
<code>
blog = "/news/:title/"
</code>

Correct?

Comment (Krzmbrzl) [reply hugo/config.toml:9]: I have never done that so I don't know ^^
But it should be easily testable. Just try it out, run the hugo webserver locally and check if the redirect is working as intended :point_up:

File: hugo/content/news/2010-08-17-for-the-record.md (hunk 2 above)
Comment (Krzmbrzl) [reply hugo/content/news/2010-08-17-for-the-record.md:23]: Not really. It rather seems like spaces have been replaced. In any case these unrelated changes should be removed from this PR.

File: hugo/content/security/Mumble-SA-2014-005.md (hunk 3 above)
Comment (Krzmbrzl) [reply hugo/content/security/Mumble-SA-2014-005.md:1]: The entire file seems to have changed but the content seems to be pretty much the same still. So the question is: What are these changes about?

File: hugo/config.toml (hunk 1 above)
Comment (toby63) [reply hugo/config.toml:9]: I don't see instructions for that.
So I will surely not look through all of this to find out how to build it exactly.

File: hugo/content/news/2010-08-17-for-the-record.md (hunk 2 above)
Comment (toby63) [reply hugo/content/news/2010-08-17-for-the-record.md:23]: Well, I will see if I have time and interest to check all the files for that.
I use a completely normal program (ghostwriter), so I don't know what the reasons for this are exactly.

File: hugo/content/security/Mumble-SA-2014-005.md (hunk 3 above)
Comment (toby63) [reply hugo/content/security/Mumble-SA-2014-005.md:1]: I have done nothing.
I only opened the file, changed the parts (mostly exchanged blog for news) and saved it.
I don't know why git thinks it must replace everything.

Comment (Krzmbrzl) [reply hugo/content/security/Mumble-SA-2014-005.md:1]: The chances of "git thinking" this are pretty slim I would say. I rather assume that your text editor did some sort of transformation automatically and thus the files you edited changed in multiple ways.

Comment (toby63) [reply hugo/content/security/Mumble-SA-2014-005.md:1]: You are probably correct with that.
I will see if I can use an alternative in the future.

I saw that not so much has gone wrong, so I hope to be able to fix it.

**Comment by toby63** [comment]
Should be better now :thinking:.

File: hugo/config.toml [hunk 4]
```diff
@@ -6,7 +6,8 @@ publishDir = "../public"
...
+  news = "/news/:title/"
+  blog = "/news/:title/"
```
Comment (Kissaki) [inline hugo/config.toml:10]: I presume this sets up a duplicate system rather than a new one with the old one redirecting?

I think blog should be an alias. https://gohugo.io/content-management/urls/

**Comment by Kissaki** [comment]
What’s the reasoning for the change? Neither the commit description nor the PR description gives context.

File: hugo/content/downloads.md [hunk 5]
```diff
@@ -120,7 +120,7 @@ We do not currently provide an official Android client. However, you can use one
 Development snapshots contain unreleased features and changes that will eventually be available in the next stable release. Please report any problems you encounter on [our issue tracker](https://github.com/mumble-voip/mumble/issues).
 
 The most recent snapshot version is the first snapshot of Mumble version 1.4.0
-([Release Announcement](https://www.mumble.info/blog/first-mumble-1.4.0-development-snapshot/)).
+([Release Announcement](https://www.mumble.info/news/first-mumble-1.4.0-development-snapshot/)).
```
Comment (Kissaki) [inline hugo/content/downloads.md:123]: This is unrelated to this change, but this link should be relative, not absolute; at the very least in source.

**Comment by Krzmbrzl** [comment]
> Will still need a test though; I can't see how it's done exactly, so I think someone should write a little guide how to do it.
I only see a .bat file for windows.

Install hugo on your local system, cd into the `hugo` dir and run `hugo server`

File: hugo/content/downloads.md (hunk 5 above)
Comment (toby63) [reply hugo/content/downloads.md:123]: You mean these `relref` tags?
While I agree that relative linking is better, I also think that a simple syntax is very good.
So if it is possible it should work with something like: `(news/first-mumble-1.4.0-development-snapshot/)`

File: hugo/config.toml (hunk 4 above)
Comment (toby63) [reply hugo/config.toml:10]: > question: I presume this sets up a duplicate system rather than a new one with the old one redirecting?

I don't know if I understand your question correctly, but:
- It is intended to use `/news` only, so no duplicate system. I have also changed all links on the website to `/news`.
- this `blog = "/news/:title/"` is only intended to redirect as suggested by @Krzmbrzl 

If you prefer other methods for a general redirect, you are free to suggest or implement them.

> I think blog should be an alias. https://gohugo.io/content-management/urls/

Regarding the negative answer above, this is not the ideal solution.

**Comment by toby63** [comment]
@Kissaki 
> What’s the reason and reasoning for the change? Neither the commit description nor the PR description gives context.

I can add a reason, but I thought it is obvious.
Most websites call it `News`, so I thought for better overview it should be renamed.

> Install hugo on your local system, cd into the hugo dir and run hugo server

If thats the case, then it should maybe be added to README.md.

**Comment by Krzmbrzl** [comment]
Yes adding that to the README is a good idea. I just never came around doing it

File: hugo/content/downloads.md (hunk 5 above)
Comment (Kissaki) [reply hugo/content/downloads.md:123]: That’s outside our control. We will have to use the hugo mechanism, which is the rel functions. We should not introduce an additional syntax or mechnism, which would wrap that mechanism. The ref syntax is the _simple syntax_. (The alternative would be introducing a shortcode or functions.)

**Comment by Kissaki** [comment]
There is a semantic difference between blog and news. Those are distinctive things and imply different things. They are closely related, sometimes similar, but not the same.

Development and community projects often use _blog_ rather than _news_.

**Comment by Krzmbrzl** [comment]
To be fair though, we only use it for announcing _News_ and we don't use it as a _Blog_ :shrug:

File: hugo/content/news/2014-03-12-mumble-for-ios-1-3-0.md [hunk 6]
```diff
@@ -17,4 +17,4 @@ As always, it’s available on the [App Store][1]. And don’t forget — if you
 The Mumble team
 
  [1]: https://itunes.apple.com/us/app/mumble/id443472808?mt=8
- [2]: http://blog.mumble.info/mumble-for-ios-1-2-3-is-buggy-when-running-on-ios-7-1/
+ [2]: https://www.mumble.info/news/mumble-for-ios-1.2.3-is-buggy-when-running-on-ios-7.1/
```
Comment (Krzmbrzl) [inline hugo/content/news/2014-03-12-mumble-for-ios-1-3-0.md:20]: Are you sure that the link has to be adjusted to use dots instead of dashes? :thinking:

Comment (toby63) [reply hugo/content/news/2014-03-12-mumble-for-ios-1-3-0.md:20]: I have simply copied the link directly from the browser and changed `blog` to `news`, so I assume: yes that is correct and necessary.

Comment (Krzmbrzl) [reply hugo/content/news/2014-03-12-mumble-for-ios-1-3-0.md:20]: Just checked: So the old link is also working but it seems to be a redirect to the one using the actual dots :point_up:

**Review by Krzmbrzl — APPROVED** [review]

File: hugo/content/downloads.md (hunk 5 above)
Comment (toby63) [reply hugo/content/downloads.md:123]: Found this: https://github.com/13rac1/goldmark-wikilink

Sadly upstream hugo does not seem to implement it by themselves: https://github.com/gohugoio/hugo/issues/3606

**Review by Kissaki — CHANGES_REQUESTED** [review]
Request: Please add a description with reasoning to the PR description, and preferably also the commit message.

While not strictly necessary, the Git commits should document changes, and why they happen.

It’s important for reviewing to get on the same page, and provide a reason for the change, which would otherwise not be necessary. And in the future, it is important for documenting the change when looking back at it.

You offered to add it in a comment, but did not do so yet. If you do not want to, and explicitly say so, I guess it’s not a blocker.

---

Thought: It would have been nice if the changes were made in two commits: One handling the renames, and another making code changes like DOM IDs and related CSS rules. The mixture of the two makes it unnecessarily hard to review, look at and look for logical changes.

I would split the commits. If you do not want to do that, it’s not a blocker.

---

Blocker: When I open the website and access /blog/ it is not a redirect but still accessible as such. This means pages are duplicated, which is an issue for SEO and clarity.

File: hugo/config.toml (hunk 4 above)
Comment (Kissaki) [reply hugo/config.toml:10]: I just tested it and as I suspected /blog/ is still accessible as pages. It is not redirects to /news/.

File: hugo/content/downloads.md (hunk 5 above)
Comment (Kissaki) [reply hugo/content/downloads.md:123]: I fixed this in master. Unfortunately this will result in a conflict on this branch/PR. But given that it’s a small change, and this PR still has open points, I decided to go ahead and fix it.

**Comment by Kissaki** [comment]
> Yes adding that to the README is a good idea. I just never came around doing it

I extended the README with more information.

File: hugo/config.toml (hunk 4 above)
Comment (toby63) [reply hugo/config.toml:10]: That is bad. I guess we need a different solution then.

**Comment by toby63** [comment]
1. 
> Request: Please add a description with reasoning to the PR description, and preferably also the commit message.

I will do that in some time, but not today.
Still I think in this case it is obvious, so I don't know why that in necessary, it's a simple name change that most people should know what it means.

2. 
> Thought: It would have been nice if the changes were made in two commits: One handling the renames, and another making code changes like DOM IDs and related CSS rules. The mixture of the two makes it unnecessarily hard to review, look at and look for logical changes.

Well honestly said: I will take a second look on this, but if these are too many changes I will probably not manually split that.

3.
> Blocker: When I open the website and access /blog/ it is not a redirect but still accessible as such. This means pages are duplicated, which is an issue for SEO and clarity.

Well I don't know a solution for that.
As this is your website and your website system, it might be up to you to suggest a solution.
Ideas that come to my mind are: 
- Redirect option in the provider menu
- Redirect with .htaccess or similar
- some more options in hugo

Only thing I can say is: I will not add redirects to any single blog page (because: too much work), if you want that solution, you have to do it.

**Comment by Kissaki** [comment]
> 
> 
> 
> > Request: Please add a description with reasoning to the PR description, and preferably also the commit message.
> 
> I will do that in some time, but not today.
> Still I think in this case it is obvious, so I don't know why that in necessary, it's a simple name change that most people should know what it means.
> 
> 
> > Thought: It would have been nice if the changes were made in two commits: One handling the renames, and another making code changes like DOM IDs and related CSS rules. The mixture of the two makes it unnecessarily hard to review, look at and look for logical changes.
> 
> Well honestly said: I will take a second look on this, but if these are too many changes I will probably not manually split that.
> 
> 
> > Blocker: When I open the website and access /blog/ it is not a redirect but still accessible as such. This means pages are duplicated, which is an issue for SEO and clarity.
> 
> Well I don't know a solution for that.
> As this is your website and your website system, it might be up to you to suggest a solution.
> Ideas that come to my mind are:
> 
>     * Redirect option in the provider menu
> 
>     * Redirect with .htaccess or similar
> 
>     * some more options in hugo
> 
> 
> Only thing I can say is: I will not add redirects to any single blog page (because: too much work), if you want that solution, you have to do it.

If you think it’s obvious you should be able to argue for it.

I provided arguments for why it is important to do so. You agreed to add a description, which did not happen yet.

You said it seemed obvious before. You repeating it like that, in those words, makes me feel as if you called me stupid.

I provided information on how they are different, and why it is important to describe the change. All you do is repeat that it is obvious.

You should always assume the least amount of knowledge on other parties. Otherwise it is documentation for yourself right now. It is not self explanatory or descriptive, or useful to others, or possibly not even for yourself in two years.

**Comment by Kissaki** [comment]
> > Blocker: When I open the website and access /blog/ it is not a redirect but still accessible as such. This means pages are duplicated, which is an issue for SEO and clarity.
> 
> Well I don't know a solution for that.
> As this is your website and your website system, it might be up to you to suggest a solution.
> Ideas that come to my mind are:
> 
>     * Redirect option in the provider menu
> 
>     * Redirect with .htaccess or similar
> 
>     * some more options in hugo
> 
> 
> Only thing I can say is: I will not add redirects to any single blog page (because: too much work), if you want that solution, you have to do it.

I linked aliases above, which I think are for this case. Additional go webserver redirect setups may be viable too though.

Either way I am against introducing a duplication of the content under a different name, where only the new name is linked but the old content is still readable and indexed. It is not a source rename but content duplication like this.

**Comment by toby63** [comment]
**Description of commit:** 
As I said I will add it at some point in the near future, but not now, because I have better things to do right now.
Regarding the topic: As I have said, this is a simple name change from "blog" to "news" and what both things mean and that they are related should be obvious imo.
So the only potential for further explanation I see are:
- specify that the name change is for the page (and folder) "blog" which will be renamed
- and that I/we choose "news", because that is better suited for the content, as it is mostly news about new releases and it is better recognizable, cause most other websites use "news" too.

> I linked aliases above, which I think are for this case. 

I might be wrong about this (as I might not understand the documentation correctly), but it seems to me that aliases are to be added to each blog post (page) individually, which is too much work for this imo.

> Additional go webserver redirect setups may be viable too though.

Sounds good.

> Either way I am against introducing a duplication of the content under a different name, where only the new name is linked but the old content is still readable and indexed. It is not a source rename but content duplication like this.

I don't understand, what is the alternative in your oppinion (aside from no name change obviously)?

**Comment by Kissaki** [comment]
>     * and that I/we choose "news", because that is better suited for the content, as it is mostly news about new releases and it is better recognizable, cause most other websites use "news" too.

This is a valid reason. If we use it basically only for news, and not blog posts, then this change makes sense.

I guess this is something we should discuss in a team meeting, whether we want to write blog posts too (I would like to but evidently do not have the time for it), or not and as a consequence rename it to news.

**Comment by Kissaki** [comment]
> > Either way I am against introducing a duplication of the content under a different name, where only the new name is linked but the old content is still readable and indexed. It is not a source rename but content duplication like this.
> 
> I don't understand, what is the alternative in your oppinion (aside from no name change obviously)?

The old URLs have to redirect to the corresponding new URL, which hosts the same content.

**Comment by toby63** [comment]
> This is a valid reason. If we use it basically only for news, and not blog posts, then this change makes sense.
> I guess this is something we should discuss in a team meeting, whether we want to write blog posts too (I would like to but evidently do not have the time for it), or not and as a consequence rename it to news.

Ok, then I will wait for that decision. I still would choose the name "news".

> The old URLs have to redirect to the corresponding new URL, which hosts the same content under the new URL.

That was the plan all along. I guess we just misunderstood each other regarding your comment about duplication.

**Comment by Kissaki** [comment]
We discussed this yesterday in our team meeting.

We concluded that we do not want to change it from Blog to News.

We have old technical posts and wish for or can imagine non-news announcement posts in the future, even if we have not posted any recently. We want to prevent moving and invalidating all old URLs, especially when we may want to revert this change once we do more of those other kind of posts.

We also discussed an alternative of keeping URLs but renaming the menu item.

We want to make news more obvious and discoverable. And especially so for users who may not know what a 'Blog' is. We want to add a 'Latest News' block to the home page for discoverability. I offered to do this, we assigned this to me. I see now we already have #141 which mentions this, or something similar.

We also have opportunity for some structuring between news and other kinds of posts, but for that no immediate or reasonably important need or value right now. An idea that may be worth to revisit in the future though.

As a retrospective and feedback, in case it will be valued:

It is hard to read, understand and discuss things when we do not have a common understanding of the problem space, arguments, weighing of those arguments and consequences. A PR description should establish all of those things; describe the problem space, establish a common understanding, arguments for challenges and solutions, and it should include negatives that are deemed acceptable too.

This PR description did not do that. Which made it very difficult to argue about, and unnecessarily inefficient, wasting time to potentially find this common understanding with back and forths.

So it would be great if that were different in the future.

Thanks for the underlying suggestion though. :) We were able to discuss it now, in the end, and make a focused decision and good plan/compromise for it now in our meeting. And we can continue this in #141.

**Comment by toby63** [comment]
Ok.

> We want to make news more obvious and discoverable. And especially so for users who may not know what a 'Blog' is. We want to add a 'Latest News' block to the home page for discoverability. I offered to do this, we assigned this to me. I see now we already have #141 which mentions this, or something similar.

Sounds good :+1:.

> And especially so for users who may not know what a 'Blog' is.

It is not about people not knowing what a blog is, probably everyone knows that.
It is about what people expect from a page named "blog".
Often blogs feature very lengthy texts about detailed (in this case) technical stuff, so it might not be of interest.
While at the same time classical project news are interesting to most users (releases, important problems, important project changes etc.).

> It is hard to read, understand and discuss things when we do not have a common understanding of the problem space, arguments, weighing of those arguments and consequences. 

In regular I agree with this, but:
1. In this case I thought everything is pretty obvious. It is about a name change, the title says it exactly. It is no complicated topic and no technical names are used that would need explanation.
2. If you want something like this for every PR, you might add it to the commit guidelines etc.
